### PR TITLE
Support directly handling enum and int

### PR DESCRIPTION
**Pull request by danielmarbach**
@SimonCropp and I decided to get rid of `LegacyEnumResponse`. It has no meaning at all.

We now support to handle `int` and `enum` directly by doing the following:

----
Sender side:
`bus.RequestWithTransientlyHandledResponse<OldEnum>(new MyRequest(), new SendOptions());`

Receiver side:
`bus.Reply(OldEnum.Success);`

----

or

----
Sender side:
`bus.RequestWithTransientlyHandledResponse<int>(new MyRequest(), new SendOptions());`

Receiver side:
`bus.Reply(200);`

----

this makes the API much more convenient to use. `int` and `enum` are added to the system message conventions. Therefore it is not possible to just do (using `string` as an example)

`bus.Reply("somestring")`

Furthermore, this PR adds more acceptance tests for the different variations of the API.

### CODE REVIEW

**Comment by danielmarbach** [comment]
@SimonCropp I stopped using the word `Legacy`. Please review. I kept the original ATT namings. One potential good addition could be to align the ATT names more with the terms you came up in the doco PR. Feel free to do those changes if necessary and merge if good to go.

**Comment by johnsimons** [comment]
@danielmarbach I find this very similar to what the `bus.Return` used to be.
IMO, if you guys think that this should be supported then I would bring back the `bus.Return`. the reason is because by providing samples/doco with `bus.Reply(OldEnum.Success);` or `bus.Reply(200);`, it may be sending the wrong message to users that it is ok to send `enums/ints`.
So in summary could this be fuelling an anti-pattern ?

**Comment by SimonCropp** [comment]
@johnsimons while we want to convey that this is an anti-pattern forcing a user to acknowledge this on every single use through unnecessary verbosity is not a good experience for them.

If this is really an anti-pattern and we want to eventually not support it then we should go back to `bus.Return` and have a proper obsolete with warning as to when we want to delete.

Also i thought the act of moving this whole functionality outside the core, and associated doco around why they are seperate, already converys the anti-pattern.

So in summary i like this PR although i would consider going back to `bus.Return`.

**Comment by danielmarbach** [comment]
What I like about Reply is that we have now one single entry point to reply.

File: src/NServiceBus.Callbacks.AcceptanceTests/When_a_callback_for_local_message_canceled.cs [hunk 1]
```diff
@@ -13,11 +13,13 @@ public class When_a_callback_for_local_message_canceled : NServiceBusAcceptanceT
...
-            Scenario.Define<Context>()
-                    .WithEndpoint<EndpointWithLocalCallback>(b => b.Given(async (bus, context) =>
+            var context = new Context();
+
+            Scenario.Define(context)
+                    .WithEndpoint<EndpointWithLocalCallback>(b => b.Given(async (bus, ctx) =>
                         {
                             var cs = new CancellationTokenSource();
-                            context.TokenSource = cs;
+                            ctx.TokenSource = cs;
```
Comment (SimonCropp) [inline src/NServiceBus.Callbacks.AcceptanceTests/When_a_callback_for_local_message_canceled.cs]: why move to a variable name that means nothing?

**Comment by SimonCropp** [comment]
i am happy for this to be merged so we can get the docs + samples all working. we can discus the exact method names in another PR.
